C++/4. Advanced-Operators.md

```diff
@@ -116,11 +120,13 @@ Logical Operators are the operators which are used when comparing one more thing
 - `||`
 - `!`
 
-`&&` means and. \
-`||` means or. \
-`!` means not.
+`&&` means "and." \
+`||` means "or." \
+`!` means "not."
+
+The above operators are used with two Relational Operator statements or with two of other things. Like if you have `4 > 5 || 4 != 5` the result will be `true` because at least one of the two statements are true (`4 != 5`) and we said `||` so only one of them have to be true. `&&` means both / all of them have to be true and `!` means the result is reversed (`true` => `false` and `false` => `true`).
 
-The above operators are used with two Relational Operator statements or with two of other things. Like if you have `4 > 5 || 4 != 5` the result will be `1` because `1` stands for true and at least one of the two statements are true (`4 != 5`) and we said `||` so only one of them have to be true. `&&` means both / all of them have to be true and `!` means the result is reversed (`true` = `false` and `false` = `true`).
+**Note: In English, normally when you say that you're going to a store ___or___ a concert, you mean that you are intending to do only of the things, not both; C++'s `||` operator returns `true` for `true || true`, but C++ doesn't provide an operator that would yield `false` for `true || true`. (there is a way to captures the former meaning though, and will be covered later)**
```

Comment:
Does anyone disagree with this comment, or feel that it is out of place?

Reply:
i disagree! You can use the `!` operator

Reply:
I'm referring to an operation that can generate this truth table:

|    x    |    y    | result |
| :----- | :----- | :----- |
| false | false  | false |
| false | true   | true  |
| true  | false  | true  |
| true  | true   | false |

The equivalent could be:
```c++
bool _xor(
  bool const x,
  bool const y
) {
  return x && y
    ? false
    : x || y;
}
```
or it could be:
```c++
bool _xor(
  bool const x,
  bool const y
) {
  return 0u == (static_cast<unsigned>(x) ^ static_cast<unsigned>(y));
}
```

as xor (`^`) has this truth table, but implicitly upcasts booleans to numbers. (<3 Rust just returns a bool, allowing `true ^ false`)

Reply:
I didn't understand the text but I understood the code